zapcore/write_syncer_test.go

```diff
@@ -65,6 +70,80 @@ func TestAddSyncWriter(t *testing.T) {
 	assert.NoError(t, ws.Sync(), "Unexpected error calling a no-op Sync method.")
 }
 
+func TestBufferWriter(t *testing.T) {
+	// If we pass a plain io.Writer, make sure that we still get a WriteSyncer
+	// with a no-op Sync.
+	t.Run("sync", func(t *testing.T) {
+		buf := &bytes.Buffer{}
+		ws, close := Buffer(AddSync(buf), 0, 0)
+		defer close()
+		requireWriteWorks(t, ws)
+		assert.Equal(t, "", buf.String(), "Unexpected log calling a no-op Write method.")
```

Comment:
We can use `assert.Empty` here:

```suggestion
		assert.Empty(t, buf.String(), "Unexpected log calling a no-op Write method.")
```